playground/src/lib.rs

```diff
@@ -0,0 +1,209 @@
+use fancy_regex::internal::{FLAG_CASEI, FLAG_DOTNL, FLAG_IGNORE_SPACE, FLAG_MULTI, FLAG_UNICODE};
+use fancy_regex::{Regex, RegexBuilder};
+use serde::{Deserialize, Serialize};
+use wasm_bindgen::prelude::*;
+
+// Expose console.log for debugging
+#[wasm_bindgen]
+extern "C" {
+    #[wasm_bindgen(js_namespace = console)]
+    fn log(s: &str);
+}
+
+// Macro for console.log
+macro_rules! console_log {
+    ($($t:tt)*) => (log(&format_args!($($t)*).to_string()))
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct Match {
+    pub start: usize,
+    pub end: usize,
+    pub text: String,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct CaptureGroup {
+    pub index: usize,
+    pub name: Option<String>,
+    pub start: Option<usize>,
+    pub end: Option<usize>,
+    pub text: Option<String>,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct MatchResult {
+    pub full_match: Option<Match>,
+    pub captures: Vec<CaptureGroup>,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct RegexFlags {
+    pub case_insensitive: bool,
+    pub multi_line: bool,
+    pub dot_matches_new_line: bool,
+    pub ignore_whitespace: bool,
+    pub unicode: bool,
+}
+
+impl Default for RegexFlags {
+    fn default() -> Self {
+        Self {
+            case_insensitive: false,
+            multi_line: false,
+            dot_matches_new_line: false,
+            ignore_whitespace: false,
+            unicode: true,
+        }
+    }
+}
+
+// Helper function to deserialize flags or use default
+fn get_flags(flags: JsValue) -> Result<RegexFlags, JsValue> {
```

Comment:
For these helpers, does the err type need to be `JsValue`, or could we just make it a `String` (it looks like bindgen handles String already?)